packages/material-ui-unstyled/src/utils/useResizeObserver.ts

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+
+interface Rect {
+  width: number;
+  height: number;
+}
+
+export default function useResizeObserver(): [Rect | null, (elem: Element) => () => void] {
+  const [rect, setRect] = React.useState<Rect | null>(null);
+  const ref = React.useCallback((elem: Element) => {
```

Comment:
How would this be used? We can't just pass it to a `ref` prop since  a `ref` prop does not support cleanup.

Reply:
I removed `useCallback` and tried this way: https://github.com/mui-org/material-ui/commit/17ec83da02895c2461d326c1c8d045ec6b2ca3ac

Let me know!